source/extensions/transport_sockets/tls/ssl_socket.cc

```diff
@@ -150,6 +161,37 @@ Network::IoResult SslSocket::doRead(Buffer::Instance& read_buffer) {
   return {action, bytes_read, end_stream};
 }
 
+void SslSocket::onPrivateKeyMethodComplete() {
+  ASSERT(isThreadSafe());
+  ASSERT(async_handshake_in_progress_);
+  async_handshake_in_progress_ = false;
+
+  ENVOY_CONN_LOG(debug, "async handshake complete", callbacks_->connection());
+  if (!handshake_complete_) {
```

Comment:
I don't think we could ever reach this codepath with `handshake_complete_ == true`.